### PR TITLE
LOB-187-account-transaction-flow

### PR DESCRIPTION
**Pull request by alinolteanufr**
LOB-187-account-transaction-flow-v4

- Refactored packages based on datamodel changes;
- Implemented support for account and transaction flows v4.

### CODE REVIEW

File: secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-datamodel/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/datamodel/account/v4_0_0/AccountAccessConsent.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.BaseConsent;
+import org.springframework.validation.annotation.Validated;
+
+import java.util.List;
+
+/**
+ * OBIE Account Access Consent: https://openbankinguk.github.io/read-write-api-site3/v3.1.10/resources-and-data-models/aisp/account-access-consents.html
+ */
+@Validated
+public class AccountAccessConsent extends BaseConsent<FRReadConsent> {
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-datamodel/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/datamodel/account/v4_0_0/AccountAccessConsent.java:28]: It doesn't look like anything is using the new classes in this pkg?

File: secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-api/src/test/java/com/forgerock/sapi/gateway/rcs/consent/store/api/account/AccountAccessConsentValidationHelpers.java [hunk 2]
```diff
@@ -36,6 +35,20 @@ public class AccountAccessConsentValidationHelpers {
...
+        assertThat(consent.getRequestVersion()).isEqualTo(obVersion);
+        assertThat(consent.getResourceOwnerId()).isNull();
+        assertThat(consent.getAuthorisedAccountIds()).isNull();
+
+        assertThat(consent.getCreationDateTime()).isBefore(new Date());
+        assertThat(consent.getStatusUpdateDateTime()).isEqualTo(consent.getCreationDateTime());
+    }
+
     public static void validateCreateConsentAgainstCreateRequest(AccountAccessConsent consent,
                                                                  CreateAccountAccessConsentRequest createAccountAccessConsentRequest) {
```
Comment (dbadham-fr) [inline secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-api/src/test/java/com/forgerock/sapi/gateway/rcs/consent/store/api/account/AccountAccessConsentValidationHelpers.java:53]: Looks like the change above is just adding a version param to this method and it looks like the original method is no longer used so it should be removed.

**Review by dbadham-fr — COMMENTED** [review]
Can you add a PR description please

File: secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-datamodel/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/datamodel/account/v4_0_0/AccountAccessConsent.java (hunk 1 above)
Comment (alinolteanufr) [reply secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-datamodel/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/datamodel/account/v4_0_0/AccountAccessConsent.java:28]: Yes, I omitted to delete them

**Review by dbadham-fr — DISMISSED** [review]

**Pull request merged**